src/System.Private.Xml/src/System/Xml/Serialization/PrimitiveXmlSerializers.cs

```diff
@@ -236,11 +236,8 @@ protected override void InitCallbacks()
         }
     }
 
-#if XMLSERIALIZERGENERATOR
+#if !XMLSERIALIZERGENERATOR
     internal class XmlSerializationPrimitiveReader : Microsoft.XmlSerializer.Generator.XmlSerializationReader
```

Comment:
Does not sound right for **Microsoft**.XmlSerializer.Generator.XmlSerializationReader living inside `#if !XMLSERIALIZERGENERATOR`.